### ISSUE TITLE
Enable [title] binding for amp-iframe

### ISSUE THREAD

**Issue by cvializ** [ampproject/amphtml#17308]
Enable amp-bind `[title]` attribute for `amp-iframe` to improve accessibility. Since the `[src]` attribute is enabled, we need to allow document authors to change the `title` attribute to better annotate the changing iframed documents for screen readers.

https://www.w3.org/TR/WCAG20-TECHS/H64.html

**Comment by atl-love** [comment]
Is this one free for the taking?

**Comment by atl-love** [comment]
Perfect! Thanks for the code snippets, I'll check it out.

**Comment by cvializ** [comment]
Hi @atl-love ! how is your work on the issue progressing? If you need any assistance with fixing the bug or with managing AMP's GitHub processes just shout and we'll be glad to help

**Comment by ampprojectbot** [comment]
This issue hasn't been  updated in awhile. @cvializ Do you have any updates?

**Comment by atl-love** [comment]
I let this one slip! I can take a look early next week, unless you all want
to assign this out.

**Comment by cvializ** [comment]
For now we can unassign this issue, and if you are able to work on it next week, we can reassign it then. Thanks again!

**Comment by atl-love** [comment]
Sounds good.

**Comment by vineeth22** [comment]
@cvializ Can I take this up?

**Comment by cvializ** [comment]
Sure feel free to take it

**Comment by maxlever** [comment]
@caroqliu hi! Is there a progress update on this? Any way I can help?

**Comment by caroqliu** [comment]
Hi @maxlever, appreciate your enthusiasm! If you'd like to contribute the fix for this, I'd be happy to review. Let me know.

**Comment by maxlever** [comment]
@caroqliu i submitted a PR. However I think the correct solution was the one originally posted by @verificatorrus. amp-img does propagate the `title` attribute, as seen here: https://github.com/ampproject/amphtml/blob/a56297bf739b03c125903ec83d1ec7839b1cc8ae/builtins/amp-img.js#L35-L37. We should either also propagate amp-iframe's title, or we should remove the propagated amp-img title, for consistency.

@cvializ is right about the HTML spec saying that the title is taken from the closest ancestor with a title. However, Lighthouse doesn't think so. It thinks there's an error for any amp page with an amp-iframe because the actual `<iframe>` doesn't have an explicitly defined `title`, and for my purposes I'd like to think LH as the source of truth here... if I'm wrong about that then this issue in LH should address this https://github.com/GoogleChrome/lighthouse/issues/10406.

**Comment by caroqliu** [comment]
Thanks @maxlever for your investigation here. I agree that the `title` should be propagated in both cases, as well as that [that change alone does not suffice](https://github.com/ampproject/amphtml/pull/17841#issuecomment-438061387) to fix this change. In order to enable binding on the `[title]` attribute on amp-iframe, we will want to use the `mutatedAttributesCallback` as described [here](https://github.com/ampproject/amphtml/issues/17308#issuecomment-426399152) _and_ propagate the mutated value as needed.

**Comment by maxlever** [comment]
thanks @caroqliu ! I have [updated my PR to reflect this](https://github.com/ampproject/amphtml/pull/27037) -- lmk if that's what you were thinking. Was not sure if we should propagate all `ATTRIBUTES_TO_PROPAGATE` or only `title`.

also, update on the HTML spec thing, I have been told in my axe-core issue that [the `title` is "relevant" but NOT inherited](https://github.com/dequelabs/axe-core/issues/2071#issuecomment-593515200), from an a11y / browser standpoint. 
> The HTML spec says the title is "relevant", but not that it is inherited. This is not how accessible name computation works. You can see this if you look up the accessible name of the iframe element (using Chrome DevTools for example).

So we need to propagate the `title` for sure.

**Comment by jwinkworth** [comment]
Can someone confirm if this has been released yet to production yet? I am encountering the same issue, and I am curious if I should be re-opening a new ticket?

Does P2:Soon mean that the issue has not yet been released?

Thanks
